doc/whats-new.rst

```diff
@@ -15,11 +15,45 @@ What's New
     np.random.seed(123456)
 
 
-.. _whats-new.2023.05.1:
+.. _whats-new.2023.06.1:
 
-v2023.05.1 (unreleased)
+v2023.06.0 (unreleased)
```

Comment:
Usually we do this in a separate commit so RTD (which points to the latest tag) doesn't show the WIP changelog